File: features/inbox/ts/Inbox.ts
import { int64 } from "./lib/less";

/**
 * 收件箱
 * @type db
 */
export class Inbox {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 收件类型
     * @index desc
     */
    type: int64 = 0

    /**
     * 接受者ID
     * @index desc
     */
    uid: int64 = 0

    /**
     * 发布者ID
     * @index desc
     */
    fuid: int64 = 0

    /**
     * 内容ID
     * @index desc
     */
    mid: int64 = 0

    /**
     * 内容项ID
     * @index desc
     */
    iid: int64 = 0

    /**
    * 其他数据
    * @length -1
    */
    options: any

    /**
     * 创建时间
     * @index desc
     */
    ctime: int64 = 0

}

File: features/inbox/inbox/Inbox.go
package inbox

import (
	"github.com/hailongz/golang/db"
)

type Inbox struct {
	db.Object
	Type	int64	`json:"type" name:"type" title:"收件类型" index:"desc"`
	Uid	int64	`json:"uid" name:"uid" title:"接受者ID" index:"desc"`
	Fuid	int64	`json:"fuid" name:"fuid" title:"发布者ID" index:"desc"`
	Mid	int64	`json:"mid" name:"mid" title:"内容ID" index:"desc"`
	Iid	int64	`json:"iid" name:"iid" title:"内容项ID" index:"desc"`
	Options	interface{}	`json:"options,omitempty" name:"options" title:"其他数据" length:"-1"`
	Ctime	int64	`json:"ctime" name:"ctime" title:"创建时间" index:"desc"`
}

func (O *Inbox) GetName() string {
	return "inbox"
}

func (O *Inbox) GetTitle() string {
	return "收件箱"
}

